### ISSUE TITLE
docs: setup ghpages CNAME

### ISSUE THREAD

**Issue by mauro3** [JuliaLang/julialang.github.com#463]
The new manual-system got merged: https://github.com/JuliaLang/julia/pull/18588 . Shouldn't julialang.org link to that now?  The links are in [this](https://github.com/JuliaLang/julia/pull/18588#issuecomment-265597437) post.

**Comment by MichaelHatherly** [comment]
The docs need to stay at the current address (`docs.julialang.org`) otherwise every link to them from anywhere online will break, so nothing should be change here.

We just need to set up the ghpages CNAME as mentioned in https://github.com/JuliaLang/julia/pull/18588#issuecomment-266180377, but that can only be done by someone with access to wherever the domain was registered.

**Comment by ViralBShah** [comment]
What exactly needs to be done?  I looked around a bit and tried a few things, but it was not clear to me. Note that in the github julialang organization settings, it says:
<img width="758" alt="screen shot 2016-12-13 at 11 36 18 pm" src="https://cloud.githubusercontent.com/assets/744411/21152604/020f27cc-c18d-11e6-9137-bde2492c0906.png">

**Comment by MichaelHatherly** [comment]
I think what we need is similar to how the `pkg.julialang.org` subdomain is hosted on https://github.com/JuliaCI/pkg.julialang.org. We just need to replicate that for `docs.julialang.org` and https://github.com/JuliaLang/julia.

**Comment by ViralBShah** [comment]
pkg.julialang.org is easy because it is a different github org. I don't know how to have one project page in the github org (website) serve from master, and another one in a github project (docs) serve from gh-pages.

cc @keno who can help set this up as well and may know this better in any case.

**Comment by ViralBShah** [comment]
Currently, trying to see the documention from `julialang.github.io/julia/latest` doesn't work because it tries to redirect to julialang.org and fails with a 404.

**Comment by StefanKarpinski** [comment]
It's non-trivial to even find which page describes what to do:

https://help.github.com/articles/setting-up-a-www-subdomain/

tl;dr: add CNAME record for the docs.julialang.org subdomain pointing at julialang.github.io.

**Comment by ViralBShah** [comment]
That does not work as I described above. It sends the request to the julialang.github.com repo, not this one. For example, I set up a newdocs.julialang.org CNAME, but gives a 404.

**Comment by MichaelHatherly** [comment]
> pkg.julialang.org is easy because it is a different github org.

The JunoLab site is setup like this as far as I can tell and seems to be all hosted from a single org. @MikeInnes was there anything special that needed to be done to make that work?

> julialang.github.io/julia/latest

Also needs an `en/` in the URL: http://julialang.github.io/julia/en/latest.

**Comment by ViralBShah** [comment]
Thanks. This works: http://julialang.org/julia/en/latest/

**Comment by mauro3** [comment]
But this doesn't http://docs.julialang.org/ (in case you're not aware).

**Comment by ViralBShah** [comment]
I am just mucking around with it as we speak.

**Comment by ViralBShah** [comment]
Reverted it.

**Comment by ViralBShah** [comment]
AFAICT, newdocs.julialang.org has the correct setup, but doesn't work. The CNAME points  to julialang.github.io.

**Comment by MikeInnes** [comment]
Here are the settings I have for docs.junolab:

<img width="868" alt="screenshot 2016-12-15 12 57 22" src="https://cloud.githubusercontent.com/assets/2234614/21224884/2ff822bc-c2c6-11e6-9ca3-51234610f3be.png">
<img width="751" alt="screenshot 2016-12-15 12 58 11" src="https://cloud.githubusercontent.com/assets/2234614/21224889/33dbab06-c2c6-11e6-8da0-87f3212a54c5.png">

As well as [this file](https://github.com/JunoLab/JunoDocs.jl/blob/master/CNAME) at the root of the repo. I don't think I did anything special above the instructions that github gives; bear in mind can take 24 hours for DNS changes to propagate.

**Comment by ViralBShah** [comment]
Fixed by c496ab65ea0dd112fe89a317fd5778867f106e98

**Comment by ViralBShah** [comment]
This works now: http://docs.julialang.org/en/latest/

but docs.julialang.org doesn't yet.

**Comment by ViralBShah** [comment]
@MikeInnes says we could set up redirects: 
https://help.github.com/articles/redirects-on-github-pages/

**Comment by MikeInnes** [comment]
The option is to have an index page with an HTML redirect ([e.g.](http://stackoverflow.com/questions/5411538/redirect-from-an-html-page), which may work better on second thoughts.

**Comment by MichaelHatherly** [comment]
That's what I've been doing with Documenter, https://github.com/JuliaDocs/Documenter.jl/blob/gh-pages/index.html, seems to work well. Thanks @ViralBShah, @MikeInnes for sorting this out!

**Comment by ViralBShah** [comment]
Can I just add index.html to the gh-pages branch on the toplevel with similar text?

**Comment by MichaelHatherly** [comment]
Yes, that should work. Probably best to redirect to the stable docs.

**Comment by ViralBShah** [comment]
Did that but doesn't seem to work. Is the .nojekyll file the issue?

**Comment by MichaelHatherly** [comment]
Hmm, could be. We probably don't really need `.nojekyll` anyway.

**Comment by MikeInnes** [comment]
It may take a sec for github to re-cache stuff

**Comment by MikeInnes** [comment]
aaaand it works :)

**Comment by ViralBShah** [comment]
Just deleted .nojekyll.

**Comment by ViralBShah** [comment]
How does one find other versions of the manual? Should the readthedocs be available on a different URL? 

And how do we link to the latest docs?

**Comment by MichaelHatherly** [comment]
`latest` has a version picker on the left side, but the locally built sphinx docs don't have any way of knowing about other versions; that's something that readthedocs adds in itself I believe.

We could try manually adding a version selector to the built docs for 0.3-0.5.

**Comment by ViralBShah** [comment]
Or perhaps the index.html can just let you pick any version of the manual?

**Comment by MichaelHatherly** [comment]
> Or perhaps the index.html can just let you pick any version of the manual?

That could as well. Simpler too.

**Comment by ViralBShah** [comment]
And Thank you for all this amazing work!  I just did the dns switch. :-)

**Comment by MikeInnes** [comment]
Looks like the css is no longer loading on the docs site. Could this be due to the `.nojekyll` change?

**Comment by ViralBShah** [comment]
The css seems to work for /en/latest, just not for /en/stable. Thoughts?

**Comment by ViralBShah** [comment]
Actually the version picker is ok (maybe no need to have index.html), but it doesn't seem to be present on other versions except for the latest.

**Comment by MikeInnes** [comment]
The resources for stable appear to be stored in a folder called `_static`, which could be special cased by jekyll in some way. @MichaelHatherly will likely know more about that.

**Comment by MichaelHatherly** [comment]
Yeah, I think I know what's going. One moment...

**Comment by MichaelHatherly** [comment]
Well, you're meant to be able to force the inclusion of certain folders with an `include:` (https://jekyllrb.com/docs/configuration/), but it doesn't appear to be working. I'm not too familiar with jekyll so perhaps I'm just doing something wrong.

**Comment by MichaelHatherly** [comment]
https://github.com/JuliaLang/julia/commit/770c793f6c575e259b5cbe7c5615afd735027928 = 🎉?

**Comment by mauro3** [comment]
Now stable points to the old-style docs for me (latest is still the new).

**Comment by MichaelHatherly** [comment]
> Now stable points to the old-style docs for me (latest is still the new).

The old-style docs are the "new" old docs, i.e. non-readthedocs ones, so it does seem to be working as it should now.

**Comment by mauro3** [comment]
Ah, yes, that makes sense: the new doc-system was not back-ported to 0.5, thus they are still as they always were.  However, then the problem remains that from the older docs there is no way to switch to other versions.  Only the 0.6-docs have a drop-down menu.

**Comment by ViralBShah** [comment]
I think we should backport the drop-down menu to the old versions - assuming that is not difficult to do.

**Comment by MichaelHatherly** [comment]
Yes, I'll see whether I can patch the version selector into the old docs.
